### ISSUE TITLE
Exception thrown where no Product Image file found

### ISSUE THREAD

**Issue by mcspronko** [magento/magento2#5184]
## Steps to reproduce
1. Remove image file for a product (pub/media/catalog/product directory)
2. Navigate to Admin to edit product
3. Then navigate to Category page/Product page
4. Observe the results
## Expected result
1. User is able to edit a product
2. User is able to see the product without image with default image placeholder
3. User is able to see products assigned to a category
## Actual result
1. Exception is thrown on all pages where a product without image is shown
## Magento versions

Magento 2.1 RC1, Magento 2.1 RC2, Magento 2.1 RC3, develop branch, 2.0 branch
## How to fix the issue

The absolute image path should be checked before an instance of Image is created.

```
    public function generateSwatchVariations($imageUrl)
    {
        $absoluteImagePath = $this->mediaDirectory->getAbsolutePath($this->getAttributeSwatchPath($imageUrl));
        if (!file_exists($absoluteImagePath)) {
            return $this;
        }
        foreach ($this->swatchImageTypes as $swatchType) {
            $imageConfig = $this->getImageConfig();
            $swatchNamePath = $this->generateNamePath($imageConfig, $imageUrl, $swatchType);
            $image = $this->imageFactory->create($absoluteImagePath);
            $this->setupImageProperties($image);
            $image->resize($imageConfig[$swatchType]['width'], $imageConfig[$swatchType]['height']);
            $this->setupImageProperties($image, true);
            $image->save($swatchNamePath['path_for_save'], $swatchNamePath['name']);
        }
        return $this;
    }
```

**Comment by mcspronko** [comment]
Hi @duhon,

This is definitely a bug. Why would Magento stop processing whole category page if only one product has no image file?

Thanks,
Max

**Comment by antboiko** [comment]
@mcspronko 

Hello Max, thanks for reporting this issue. We've created internal ticket MAGETWO-54718 to fix it.

Best,
Anton.

**Comment by mcspronko** [comment]
Hi @antboiko 

Thank you for internal ticket. Do you have any estimate when do you guys planning to fix this issue? It is pretty critical for Merchant to be able to add/edit a product from Admin.

Thanks,
Max

**Comment by pboisvert** [comment]
@mcspronko the team will need to rank this against all other open issues but given your help in spotting the issue it seems like we should be able to address fairly soon (perhaps next patch but need some analysis first).

**Comment by palamar** [comment]
Issue has been fixed in scope of #5497 and should to appear in develop branch at the nearest time.

**Comment by elenleonova** [comment]
The issue has been released in 2.1.3
